hibernate-rx-core/src/main/java/org/hibernate/rx/engine/impl/Cascade.java

```diff
@@ -464,13 +464,9 @@ private void cascadeToOne(
 			//not really necessary, but good for consistency...
 			final PersistenceContext persistenceContext = eventSource.getPersistenceContextInternal();
 			persistenceContext.addChildParent( child, parent );
-			try {
-				CompletionStage<?> cascade = action.cascade(eventSource, child, entityName, context, isCascadeDeleteEnabled);
-				stage = stage.thenCompose(v -> cascade).thenAccept(vv->{});
-			}
-			finally {
-				persistenceContext.removeChildParent( child );
-			}
+			stage = stage.thenCompose( v -> action.cascade( eventSource, child, entityName, context, isCascadeDeleteEnabled ) )
+					.whenComplete( (vv, e) -> persistenceContext.removeChildParent( child ) )
```

Comment:
If there is an exception and you don't rethrow it, it will dissapear

Reply:
@DavideD is that really the case with `whenComplete()`? Isn't it only `handle()` that works like that?

Reply:
Maybe you are right, I will check